src/runtime/components/Sections/admin.vue

```diff
@@ -2455,6 +2504,37 @@ const updatePageMetaData = async (seo, themeData) => {
     loading.value = false
   }
 }
+const createMedia = async (payload, external_call) => {
+
+  const fileData = payload['files[1][file]']
+  if (!payload['files[1][file]']) return
+  if (external_call !== true) {
+    loading.value = true
+  }
+  const data = new FormData()
+
+  data.append('files[1][platform_id]', '1')
+  data.append('files[1][file]', fileData)
+  data.append('type', fileData.type.includes('image') ? 'image' : 'document')
+  data.append('title', payload['title'] || '')
+  data.append('private_status', payload['private_status'] || 'public')
+  data.append('locked_status', payload['locked_status'] || 'unlocked')
+
+  const token = useCookie("sections-auth-token").value
+
+  try {
+    const res = await $fetch(`${nuxtApp.$sections.serverUrl}/project/${nuxtApp.$sections.projectId}/media/`, {
+      method: 'POST',
+      headers: sectionHeader({ token: token }),
+      body: data
+    })
+    return res
+  } catch (e) {
+    return e
+  } finally {
+    loading.value = false
+  }
+}
```

Comment:
why is this logic in this component ?

Reply:
shouldn't you then organize the code and put that in a separate file for clarity (folder name, file name should be explicit so we understand it's injected into the window object)

Reply:
These functions are used by nuxt-sections library so they have to stay there and they are exposed to allow host projects to use them when necessary, like we did of opening edit mode and showing the guide etc...

Sections CMS is just an installed that is not directly used by host projects except for the 2 CSS presets that are exposed using S3